File: mobile_thru_project_server_full/mobile_thru_project_server_full/src/main/java/com/ssafy/greenplate/model/service/ProductService.java
package com.ssafy.greenplate.model.service;

import java.util.List;
import java.util.Map;

import com.ssafy.greenplate.model.dto.Product;

public interface ProductService {
    /**
     * 모든 상품 정보를 반환한다.
     * 
     * @return
     */
    List<Product> getProductList();

    /**
     * id에 해당하는 상품 정보를 반환한다.
     * 
     * @param productId
     * @return
     */
    Product getProduct(Integer productId);

    /**
     * 주문량이 가장 많은 메뉴 3개 반환
     */
    List<Product> selectTop3Menu();
}

File: green_plate_project/app/src/main/java/com/ssafy/green_plate/api/ProductService.kt
package com.ssafy.green_plate.api

import com.ssafy.green_plate.dto.Product
import retrofit2.http.GET
import retrofit2.http.Path

interface ProductService {
    // 전체 상품의 목록을 반환한다
    @GET("rest/product")
    suspend fun getProductList(): List<Product>

//    // {productId}에 해당하는 상품의 정보를 comment와 함께 반환한다.
//    // comment 조회시 사용
//    @GET("rest/product/{productId}")
//    suspend fun getProductWithComments(@Path("productId") productId: Int): List<MenuDetailWithCommentResponse>
}